### PR TITLE
Add validation to ConfigMap patch test

### LINKED ISSUE #78832: Promote ConfigMap Tests to Conformance
Using the list of tested stable endpoints referenced in https://github.com/kubernetes/kubernetes/issues/78748, we can pick out the endpoints which are part of the ConfigMap kind:

[patchCoreV1NamespacedConfigMap](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-patchCoreV1NamespacedConfigMap&showUntested=false&showConformanceTested=false)

This is a stable core endpoint which has existing testing, potentially making for low-hanging fruit to upgrade to Conformance.

/area test
/area conformance
/sig testing

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Adds validation to e2e test
```
should patch ConfigMap successfully
```

**Which issue(s) this PR fixes**:
Fixes #78832

**Special notes for your reviewer**:
Part of Umbrella Issue #78748
This test is related to #78924

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/area test
@kubernetes/sig-apps-pr-reviews
@kubernetes/sig-architecture-pr-reviews

### CODE REVIEW

**Comment by spiffxp** [comment]
/ok-to-test

>Promotes the following E2E test to Conformance:

please edit your PR description, this adds a new test, doesn't actually promote

/cc @oomichi @spiffxp
/uncc @krmayankk

File: test/e2e/common/configmap.go [hunk 1]
```diff
@@ -134,6 +135,27 @@ var _ = ginkgo.Describe("[sig-node] ConfigMap", func() {
 		configMap, err := newConfigMapWithEmptyKey(f)
 		framework.ExpectError(err, "created configMap %q with empty key in namespace %q", configMap.Name, f.Namespace.Name)
 	})
+
+	ginkgo.It("should patch ConfigMap successfully", func() {
```
Comment (timothysc) [inline test/e2e/common/configmap.go]: this should be using the framework on this call line vs. ginkgo. 

This applies to everything here.

File: test/e2e/common/configmap.go [hunk 2]
```diff
@@ -134,6 +135,27 @@ var _ = ginkgo.Describe("[sig-node] ConfigMap", func() {
...
+		name := "configmap-test-" + string(uuid.NewUUID())
```
Comment (timothysc) [inline test/e2e/common/configmap.go]: Why do this vs. using the utility below?

**Review by timothysc — CHANGES_REQUESTED** [review]

File: test/e2e/common/configmap.go (hunk 1 above)
Comment (spiffxp) [reply test/e2e/common/configmap.go]: we don't have a `framework.It`; we do have a `framework.ConformanceIt` but explicitly disallow new tests from being added with ConformanceIt, they must be promoted into it

Comment (spiffxp) [reply test/e2e/common/configmap.go]: we don't have a `framework.It`; we do have a `framework.ConformanceIt` but explicitly disallow new tests from being added with ConformanceIt, they must be promoted into it

Comment (timothysc) [reply test/e2e/common/configmap.go]: Sorry retract, I'm brain fry, comment below still applies though.

**Review by spiffxp — CHANGES_REQUESTED** [review]
one nit

File: test/e2e/common/configmap.go [hunk 3]
```diff
@@ -134,6 +135,27 @@ var _ = ginkgo.Describe("[sig-node] ConfigMap", func() {
...
+			"data": "value",
+		}
+		ginkgo.By(fmt.Sprintf("Updating configMap %v/%v", f.Namespace.Name, configMap.Name))
+		_, err = f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Update(configMap)
+		framework.ExpectNoError(err)
+
+		configMapFromUpdate, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Get(name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		ginkgo.By(fmt.Sprintf("Verifying update of configMap %v/%v", f.Namespace.Name, configMap.Name))
+		gomega.Expect(configMapFromUpdate.Data).NotTo(gomega.Equal(configMapOriginalState.Data))
```
[the remark this answers is not in the document]
Comment (BobyMCbobs) [reply test/e2e/common/configmap.go]: That's a useful function! Thank you.
I've updated the test to use `framework.ExpectNotEqual` in b10743236ef

File: test/e2e/common/configmap.go [hunk 4]
```diff
@@ -134,6 +134,27 @@ var _ = ginkgo.Describe("[sig-node] ConfigMap", func() {
...
+			"data": "value",
+		}
+		ginkgo.By(fmt.Sprintf("Updating configMap %v/%v", f.Namespace.Name, configMap.Name))
+		_, err = f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Update(configMap)
+		framework.ExpectNoError(err)
+
+		configMapFromUpdate, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Get(name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		ginkgo.By(fmt.Sprintf("Verifying update of configMap %v/%v", f.Namespace.Name, configMap.Name))
+		framework.ExpectNotEqual(configMapFromUpdate.Data, configMapOriginalState.Data)
```
Comment (oomichi) [inline test/e2e/common/configmap.go]: Why don't we check the `Get()` data is the same as `Update()`d data(`"data": "value"`) here?
I mean it is nice to add `framework.ExpectEqual(configMapFromUpdate.Data, configMap.Data)`

**Review by oomichi — COMMENTED** [review]
Most part is good for me, just one comment for one more check.

File: test/e2e/common/configmap.go (hunk 4 above)
Comment (BobyMCbobs) [reply test/e2e/common/configmap.go]: configMap.Data's value is changed, so it doesn't make sense to compare against it.

[test/e2e/common/configmap.go: L146-148](https://github.com/ii/kubernetes/blob/b10743236ef08ab8cf4b3cf5ac4ea9911db453fa/test/e2e/common/configmap.go#L146)
```golang
 		configMap.Data = map[string]string{
			"data": "value",
		}
```

Comment (oomichi) [reply test/e2e/common/configmap.go]: I am not sure I can follow what you are saying here.
In this test, the config map is updated with
```
 		configMap.Data = map[string]string{
			"data": "value",
		}
```
So the config map date which is gotten at line 153 should be the same as the above value, isn't it?

Comment (BobyMCbobs) [reply test/e2e/common/configmap.go]: Perhaps the purpose of the variables is being confused.
`configMap` is the original configMap value.   
`configMapOriginalState` has the value of `configMap` before it's changed.  
`configMapFromUpdate` has the value of `configMap` after it's updated, pulling it from it's state in the cluster -- must be different from `configMapOriginalState`.

Comment (oomichi) [reply test/e2e/common/configmap.go]: > configMapFromUpdate has the value of configMap after it's updated, pulling it from it's state in the cluster -- must be different from configMapOriginalState.

Yes right. Then it is better to check not only difference from the original but also that the updated data should be the same as the data which is used for updating (`"data": "value"`)

Comment (BobyMCbobs) [reply test/e2e/common/configmap.go]: Updating the Expect statement in 7218182f119.

**Comment by oomichi** [comment]
@BobyMCbobs Thanks for updating

/lgtm

**Pull request merged**